spec/models/track_spec.rb

```diff
@@ -54,4 +187,226 @@
+      @an_event_of_the_track = create(:event, program: self_organized_track.program, track: self_organized_track)
+    end
+
+    it 'revokes the role of the track organizer' do
+      expect(@a_track_organizer.has_role?(:track_organizer, self_organized_track)).to eq true
+      self_organized_track.revoke_role_and_cleanup
+      expect(@a_track_organizer.has_role?(:track_organizer, self_organized_track)).to eq false
+    end
+
+    it 'removes the track from the events that have it set' do
+      expect(@an_event_of_the_track.track).to eq self_organized_track
+      self_organized_track.revoke_role_and_cleanup
+      @an_event_of_the_track.reload
+      expect(@an_event_of_the_track.track).to eq nil
+    end
+
+    it 'is executed when the track is canceled' do
+      self_organized_track.state = 'confirmed'
+      self_organized_track.save!
+      self_organized_track.cancel
+      expect(@a_track_organizer.has_role?(:track_organizer, self_organized_track)).to eq false
+      @an_event_of_the_track.reload
+      expect(@an_event_of_the_track.track).to eq nil
+    end
+
+    it 'is executed when the track is withdrawn' do
+      self_organized_track.withdraw
+      expect(@a_track_organizer.has_role?(:track_organizer, self_organized_track)).to eq false
+      @an_event_of_the_track.reload
+      expect(@an_event_of_the_track.track).to eq nil
+    end
+  end
+
+  describe '#accepted?' do
+    context 'returns true' do
+      it 'when the state is "accepted"' do
+        self_organized_track.state = 'accepted'
+        self_organized_track.save!
+        expect(self_organized_track.accepted?).to eq true
+      end
+    end
+
+    context 'returns false' do
+      %w[new to_accept confirmed to_reject rejected canceled withdrawn].each do |state|
+        it "when the state is \"#{state}\"" do
+          self_organized_track.state = state
+          self_organized_track.save!
+          expect(self_organized_track.accepted?).to eq false
+        end
+      end
+    end
+  end
+
+  describe '#confirmed?' do
+    context 'returns true' do
+      it 'when the state is "confirmed"' do
+        self_organized_track.state = 'confirmed'
+        self_organized_track.save!
+        expect(self_organized_track.confirmed?).to eq true
+      end
+    end
+
+    context 'returns false' do
+      %w[new to_accept accepted to_reject rejected canceled withdrawn].each do |state|
+        it "when the state is \"#{state}\"" do
+          self_organized_track.state = state
+          self_organized_track.save!
+          expect(self_organized_track.confirmed?).to eq false
+        end
+      end
+    end
+  end
+
+  # accepted? and confirmed? are mutually exclusive (they can't be both true)
+  describe '#self_organized_and_accepted_or_confirmed?' do
+    context 'returns true' do
+      context 'when self_organized? returns true' do
+        before :each do
+          allow(track).to receive(:self_organized?).and_return(true)
```

Comment:
If something changes in the way a self-organized track is defined and the track factory isn't updated accordingly, then this test might fail, because the `self_organized` trait doesn't satisfy the definition of a self-organized track. A failure of the factory shouldn't be a failure of this test. So, this way the test is independent from the factory